File: kotlin/PlanesAndroid/app/src/main/java/com/planes_multiplayer/android/ButtonWithPictogram.java
package com.planes_multiplayer.android;

import android.content.Context;
import android.graphics.Canvas;
import android.graphics.Color;
import android.graphics.Paint;
import androidx.appcompat.widget.AppCompatButton;
import android.util.AttributeSet;


/**
 * Parent class for all buttons with pictograms
 */
public class ButtonWithPictogram extends AppCompatButton {

    public ButtonWithPictogram(Context context) {
        super(context);
        init();
    }

    public ButtonWithPictogram(Context context, AttributeSet attrs) {
        super(context, attrs);
        init();
    }

    public ButtonWithPictogram(Context context, AttributeSet attrs, int defStyle) {
        super(context, attrs, defStyle);
        init();
    }

    public void init() {
        m_Paint = new Paint();
    }

    @Override
    public void onDraw(Canvas canvas) {
        m_Paint.setColor(Color.GRAY);
        canvas.drawRect(0, 0, getWidth(), getHeight(), m_Paint);

        m_Paint.setColor(Color.BLACK);
        CanvasPaintUtilities.drawButtonShadow(canvas, m_Paint, getWidth(), getHeight());
    }

    @Override
    protected void onMeasure(int widthMeasureSpec, int heightMeasureSpec) {
        int specSizeWidth = MeasureSpec.getSize(widthMeasureSpec);
        int specSizeHeight = MeasureSpec.getSize(heightMeasureSpec);

        int measWidth = m_MinWidth;
        int measHeight = m_MinHeight;

        if (m_MinWidth < specSizeWidth)
            measWidth = specSizeWidth;
        if (m_MinHeight < specSizeHeight)
            measHeight = specSizeHeight;

        setMeasuredDimension(measWidth, measHeight);
    }

    final int m_MinWidth = 30;
    final int m_MinHeight = 30;

    protected Paint m_Paint;
}

File: kotlin/PlanesAndroid/app/src/main/java/com/planes_multiplayer/android/ButtonWithPictogram.kt
package com.planes_multiplayer.android

import android.content.Context
import android.graphics.Canvas
import android.graphics.Color
import android.graphics.Paint
import android.util.AttributeSet
import com.planes_multiplayer.android.CanvasPaintUtilities.Functions.drawButtonShadow
import androidx.appcompat.widget.AppCompatButton
import com.planes_multiplayer.android.CanvasPaintUtilities
import android.view.View.MeasureSpec

/**
 * Parent class for all buttons with pictograms
 */
open class ButtonWithPictogram : AppCompatButton {
    val m_MinWidth = 30
    val m_MinHeight = 30
    protected var m_Paint: Paint? = null

    constructor(context: Context?) : super(context) {
        init()
    }

    constructor(context: Context?, attrs: AttributeSet?) : super(context, attrs) {
        init()
    }

    constructor(context: Context?, attrs: AttributeSet?, defStyle: Int) : super(context, attrs, defStyle) {
        init()
    }

    fun init() {
        m_Paint = Paint()
    }

    public override fun onDraw(canvas: Canvas) {
        m_Paint!!.color = Color.GRAY
        canvas.drawRect(0f, 0f, width.toFloat(), height.toFloat(), m_Paint!!)
        m_Paint!!.color = Color.BLACK
        drawButtonShadow(canvas, m_Paint, width, height)
    }

    override fun onMeasure(widthMeasureSpec: Int, heightMeasureSpec: Int) {
        val specSizeWidth = MeasureSpec.getSize(widthMeasureSpec)
        val specSizeHeight = MeasureSpec.getSize(heightMeasureSpec)
        var measWidth = m_MinWidth
        var measHeight = m_MinHeight
        if (m_MinWidth < specSizeWidth) measWidth = specSizeWidth
        if (m_MinHeight < specSizeHeight) measHeight = specSizeHeight
        setMeasuredDimension(measWidth, measHeight)
    }
}